File: 0001-0050/0014/solution.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        for i in range(len(strs[0])):
            c = strs[0][i]
            for j in range(1, len(strs)):
                if i == len(strs[j]) or strs[j][i] != c:
                    return strs[0][:i]
        return strs[0]


class Example:
    def __init__(self, strs, ans):
        self.strs = strs
        self.ans = ans


def main():
    examples = [
        Example(["flower", "flow", "flight"], "fl"),
        Example(["dog", "racecar", "car"], "")
    ]

    solve = Solution()
    for i, example in enumerate(examples):
        ans = solve.longestCommonPrefix(example.strs)
        if ans == example.ans:
            print(f"PASS: CASE {i}")
        else:
            print(f"FAIL: CASE {i}")


if __name__ == "__main__":
    main()

File: 0001-0050/0014/solution.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        for (int i = 0; i < strs[0].length(); i++) {
            char c = strs[0][i];
            for (int j = 1; j < strs.size(); j++) {
                if (i == strs[j].length() || strs[j][i] != c) {
                    return strs[0].substr(0, i);
                }
            }
        }
        return strs[0];
    }
};

struct Example {
    vector<string> strs;
    string ans;
};

int main() {
    vector<Example> examples = {
        {{"flower","flow","flight"}, "fl"},
        {{"dog","racecar","car"}, ""},
    };
    
    Solution *solve = new Solution();
    for (int i = 0; i < examples.size(); i++) {
        string ans = solve->longestCommonPrefix(examples[i].strs);
        if (ans == examples[i].ans) {
            cout << "PASS: CASE " << i << endl;
        } else {
            cout << "FAIL: CASE " << i << endl;
        }
    }
    
    return 0;
}